lib/instance.js

```diff
@@ -9,48 +9,32 @@ module.exports = function instance(input, done, progress) {
   var error = debug('nemo:instance:error');
   var startMS;
   var mocha;
-  var anyTests = false;
-  var result = {tags: input.profile.tags, total: 0, pass: 0, fail: 0};
+  var testResults = [];
   var driverConfig;
   var runner;
-  var prepareTestResult = (event, test, err) => {
+  var prepareTestResult = (test, err) => {
     if (err) {
       test.err = err.name || 'test err.name DNE';
       test.stack = err.stack || 'test err.stack DNE';
     }
     test.fullTitleString = test.fullTitle();
-    return {
-      event: 'test',
-      type: event,
-      test: CircularJSON.stringify(test),
-      tags: input.profile.tags
-    };
+    return CircularJSON.stringify(test);
   };
-  var exitInstance = function (failures, summary) {
-    // only attach this listener if we aren't running in parallel
-    if (summary) {
-      log('summary');
-      progress({
-        event: 'end',
+  var exitInstance = function (failures, _testResults) {
+    progress({
+      event: 'instance:end', payload: {
         tags: input.profile.tags,
-        duration: Date.now() - startMS
-      });
-    }
-    if (!done && process) {
-      process.on('exit', function () {
-        // exit with non-zero status if there were failures
-        console.log(`nemo exiting with ${failures || 0} failures`);
-        process.exit(failures);
-      });
-    }
-    // attempted to run with no matching suites/tests
-    if (!anyTests && done) {
-      done(summary);
+        duration: Date.now() - startMS,
+        reportFile: input.profile.reportFile,
+        testResults: _testResults
+      }
+    });
+    if (doneThread) {
```

Comment:
Can we make it so there are no cases when we don't pass in `done`? Can we just spawn a child thread even for 1 instance? I see no real issue with that, and it simplifies the execution logic.